src/main/java/nl/tudelft/ewi/devhub/server/web/resources/AssignmentsResource.java

```diff
@@ -728,4 +737,33 @@ public Assignment updateAssignment(@PathParam("courseCode") String courseCode,
 		return deliveries;
 	}
 
+	@POST
+    @Transactional
+    @Path("{assignmentId : \\d+}/distribute-tas")
+    public Response distributeTAs(@PathParam("courseCode") String courseCode,
+                                  @PathParam("editionCode") String editionCode,
+                                  @PathParam("assignmentId") long assignmentId) {
+        CourseEdition course = courses.find(courseCode, editionCode);
+        Assignment assignment = assignmentsDAO.find(course, assignmentId);
+
+        if(!(currentUser.isAdmin() || currentUser.isAssisting(course))) {
+            throw new UnauthorizedException();
+        }
+
+        List<Delivery> deliveries = deliveriesDAO.getLastDeliveries(assignment);
+
+        List<Group> groups = Lists.transform(deliveries, Delivery::getGroup);
+
+        Set<User> TAs = course.getAssistants();
+
+        groups.removeAll(Lists.transform(assignment.getAssignedTAS(), AssignedTA::getGroup));
+
+        List<AssignedTA> assignedTAS = AssignTAs.assignGroups(TAs, groups, assignment, new Random());
```

Comment:
Doesn't make a lot of sense to pass this as a parameter instead of through dependency injection, but I haven't explained DI for now as we have been focussing on other technologies for now. Passing it as a parameter still makes more sense than instantiating it in the method itself, it indicates that a Random is used and that if preferred, a more secure type of random can be provided.